submissions/MarharytaBoichenko/oop/index.js

```diff
@@ -0,0 +1,102 @@
+class Inhabitant {
+  constructor(name, species, gender, friends, saying) {
+    this.name = name;
+    this.species = species;
+    this.gender = gender;
+    this.friends = friends;
+    this.saying = saying;
+    this.properties = ["name", "species", "gender", "friends", "saying"];
+  }
+  toPrint() {
+    return this.properties
+      .filter((prop) => {
+        return Boolean(this[prop]) != false;
+      })
+      .map((prop) => `my ${prop}: ` + this[prop])
+      .join(", ");
+  }
+}
+
+class Person extends Inhabitant {
+  constructor(name, gender, friends, saying) {
+    super(name, "human", gender, friends, saying);
+    this.hands = 2;
+    this.legs = 2;
+  }
+
+  toPrint() {
+    return super.toPrint() + `, my hands: ${this.hands}, my legs: ${this.legs}`;
+  }
+}
+
+class Man extends Person {
+  constructor(name, friends, saying) {
+    super(name, "man", friends, saying);
+  }
+
+  toPrint() {
+    return super.toPrint();
+  }
+}
+
+class Woman extends Person {
+  constructor(name, friends, saying) {
+    super(name, "woman", friends, saying);
+  }
+
+  toPrint() {
+    return super.toPrint();
+  }
```

Comment:
No need to overload a method to only explicitly call an inherited version of the method.